url29ff2e53f3_nicolas3470_Settlers_of_Catan_AI_tgz-pJ8-robotGameJ8/src/soc/client/SOCHandPanel.java

```diff
@@ -0,0 +1,3030 @@
+
+        // Update the player-trading checkboxes
+        if (! playerTradingDisabled)
+        {
+            for (int i = 0; i < (game.maxPlayers - 1); i++)
+            {
+                boolean seatTaken = ! game.isSeatVacant(playerSendMap[i]);
+                playerSend[i].setBoolValue(seatTaken);
+                playerSend[i].setEnabled(seatTaken);
+                if (seatTaken)
+                {
+                    String pname = game.getPlayer(playerSendMap[i]).getName();
+                    if (pname != null)
+                        playerSend[i].setTooltipText(pname);
+                }
+            }
+        }
+    }
+
+    /**
+     * Display or update this player's trade offer, or hide if none.
+     * If a game reset request is in progress, don't show the offer, because
+     * they use the same display component ({@link #offer}).  In that case
+     * the trade offer will be refreshed after the reset is cancelled.
+     *<P>
+     * Does not display if playerIsClient.
+     */
+    public void updateCurrentOffer()
+    {
+        if (inPlay)
+        {
+            SOCTradeOffer currentOffer = player.getCurrentOffer();
+
+            if (currentOffer != null)
+            {
+                if (! (offerIsResetMessage || offerIsDiscardMessage))
+                {
+                    if (! playerIsClient)
+                    {
+                        offer.setOffer(currentOffer);
+                        offer.setVisible(true);
+                        if (offerHidesControls)
+                            hideTradeMsgShowOthers(false);
+                        offer.repaint();
+                    }
+                }
+                else
+                    offerIsMessageWasTrade = true;  // Will show after voting
+            }
+            else
+            {
+                clearOffer(false);
+            }
+        }
+    }
+
+    /**
+     * Show that this player (who isn't the client) has rejected another player's offer.
+     */
+    public void rejectOfferShowNonClient()
+    {
+        if (playerIsClient)
+            return;
+        offer.setMessage("No thanks.");
+        if (offerHidesControls)
+            hideTradeMsgShowOthers(false);
+        offer.setVisible(true);
+        //validate();
+        repaint();
+    }
+
+    /**
+     * Client is rejecting the current offer from another player.
+     * Send to the server, hide the trade message, trigger a repaint.
+     * @since 1.1.08
+     */
+    public void rejectOfferAtClient()
+    {
+        client.rejectOffer(game);
+        if (offerHidesControls)
+            hideTradeMsgShowOthers(false);
+        repaint();        
+    }
+
+    /**
+     * If the trade-offer panel is showing a message
+     * (not a trade offer), clear and hide it.
+     * Assumes this hand's player is not the client.
+     * @see #tradeSetMessage(String)
+     */
+    public void clearTradeMsg()
+    {
+        if ((offer.getMode() == TradeOfferPanel.MESSAGE_MODE)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An equality operator (== or !=) is used to compare strings, which matches strings based on address.

**Fix**: Do content comparison using the equals() method to compare the values of the strings.